src/components/member-page/donation/table.js

```diff
@@ -0,0 +1,203 @@
+  .row {
+    width: 100%;
+    display: flex;
+    gap: 8px;
+    .amount {
+      display: flex;
+      flex-grow: 1;
+      justify-content: end;
+    }
+  }
+  .donation-number {
+    width: 100%;
+    display: block;
+    word-break: break-all;
+  }
+  ${mq.tabletAndAbove`
+    display: none;
+  `}
+`
+
+const P1Gray800 = styled(P1)`
+  color: ${colorGrayscale.gray800};
+`
+
+const P1Gray600 = styled(P1)`
+  color: ${colorGrayscale.gray600};
+`
+
+const formattedDate = date =>
+  `${date.getUTCFullYear()}/${date.getUTCMonth() + 1}/${date.getUTCDate()}`
+
+const TableRow = memo(({ record }) => {
+  const {
+    type,
+    created_at: createdAt,
+    amount,
+    status,
+    order_number: orderNumber,
+  } = record
+  const donationDate = formattedDate(new Date(createdAt))
+  const amountSuffix = type === DonationType.PRIME ? '元' : '元/月'
+
+  return (
+    <React.Fragment>
+      <TableContent>
+        <P1Gray800 className="donation-date" text={donationDate} />
+        <P1Gray800 className="type" text={DonationTypeLabel[type]} />
+        <P1Gray800 className="donation-number" text={orderNumber} />
+        <P1Gray800
+          className="amount"
+          text={`${amount.toLocaleString('en-US')}${amountSuffix}`}
+        />
+        <div className="status">
+          <StatusBadge status={status} type={type} />
+        </div>
+      </TableContent>
+      <MobileTableContent>
+        <P1Gray800 text={donationDate} />
+        <div className="row">
+          <P1Gray800 weight={P1.Weight.BOLD} text={DonationTypeLabel[type]} />
+          <StatusBadge status={status} type={type} />
+          <P1Gray800
+            className="amount"
+            text={`${amount.toLocaleString('en-US')}${amountSuffix}`}
+          />
+        </div>
+        <P1Gray600 className="donation-number" text={orderNumber} />
+      </MobileTableContent>
+    </React.Fragment>
+  )
+})
+
+TableRow.propTypes = {
+  record: PropTypes.object.isRequired,
+}
+
+TableRow.displayName = 'TableRow'
+
+export const Table = ({ records }) => (
+  <TableContainer>
+    <TableHeader>
+      <P1Gray800
+        className="donation-date"
+        weight={P1.Weight.BOLD}
+        text={'贊助日期'}
+      />
+      <P1Gray800 className="type" weight={P1.Weight.BOLD} text={'項目'} />
+      <P1Gray800
+        className="donation-number"
+        weight={P1.Weight.BOLD}
+        text={'贊助編號'}
+      />
+      <P1Gray800 className="amount" weight={P1.Weight.BOLD} text={'金額'} />
+      <P1Gray800 className="status" weight={P1.Weight.BOLD} text={'狀態'} />
+      <DividerContainer>
+        <Divider />
+      </DividerContainer>
+    </TableHeader>
+    {records.map(record => (
+      <React.Fragment key={record.order_number}>
+        <TableRow record={record} />
+        <DividerContainer>
+          <Divider />
+        </DividerContainer>
+      </React.Fragment>
+    ))}
+  </TableContainer>
+)
+
+Table.propTypes = {
+  records: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
Could we add the spec of record object shape here ?

Reply:
It will be added at phase 2